forge/db/models/DeviceGroup.js

```diff
@@ -0,0 +1,109 @@
+/**
+ * A DeviceGroup.
+ * A logical grouping of devices for the primary intent of group deployments in the pipeline stages.
+ * @namespace forge.db.models.DeviceGroup
+ */
+
+const { DataTypes, literal } = require('sequelize')
+
+const { buildPaginationSearchClause } = require('../utils')
+
+module.exports = {
+    name: 'DeviceGroup',
+    schema: {
+        name: { type: DataTypes.STRING, allowNull: false },

```

Comment:
Add a [validation](https://sequelize.org/docs/v6/core-concepts/validations-and-constraints/) to this? (Untested pseudocode). It'll then throw a sensible error if you try to create a DeviceGroup without a name _before_ it hits the database (and can bubble up to the controller).


```suggestion
        name: { type: DataTypes.STRING, allowNull: false, validate: { notNull: true } },
```

Reply:
Handling example here: https://github.com/FlowFuse/flowfuse/blob/c3a5bb620862202ad484200b170ee90e10f77a0e/forge/ee/routes/pipeline/index.js#L88-L94